### ISSUE TITLE
"fopen: No such file or directory" on ER-X

### ISSUE THREAD

**Issue by julianschuh** [Lochnair/vyatta-wireguard#9]
Hello,

I'm trying to use the wireguard module on an ER-X.

```Version:      v1.9.1.1
Build ID:     4977602
Build on:     04/27/17 01:13
Copyright:    2012-2016 Ubiquiti Networks, Inc.
HW model:     EdgeRouter X 5-Port
```

wireguard version: wireguard-ralink-0.0.20170629-1.deb

The router was reset just before installing the package.

When trying to create a wireguard interface, eg. by just specifying a private key in the minimal configuration and commiting afterwards, the CLI displays the error `fopen: No such file or directory`.

```ubnt@ubnt:~$ configure
[edit]
ubnt@ubnt# set interfaces wireguard wg0 private-key xxx
[edit]
ubnt@ubnt# commit
[ interfaces wireguard wg0 private-key xxx ]
fopen: No such file or directory

Commit failed
[edit]
```

The error also happens if a completely configure the interface (with peer, address, ...)

**Comment by Lochnair** [comment]
Sorry for the late reply, I'm on vacation at the moment.

Basically you'll see this error if the private key you've set is not recognized as a a valid Base64 private key (by a regular expression), this means it'll try to open the key you've specified as a file path instead, which in all likelihood does not exist. In other words you've most likely missed a character when copy-pasting the key, or something similar.

Or there might be something funky happening with the process substitution, but that's not as likely.

**Comment by Lochnair** [comment]
@julianschuh Did you ever make any headway with this? As far as I know vyatta-wireguard doesn't work on the ER-X for anyone, and it's possible that it never has.

**Comment by Lochnair** [comment]
A kind soul on the Ubiquiti forum offered to send me an ER-X so that I can get to the bottom of this. I'll have an update for you soon after it arrives.

**Comment by sskaje** [comment]
It seems like @julianschuh run commands like 
```
set interfaces wireguard wg0 private-key oE1fQ0FXtYkMYH9eRttE0pCDeX+IRa711l8ZOm2K8lo=
```

not 
```
set interfaces wireguard wg0 private-key /config/auth/wg.private
```


I tried to set private key string as param of 'private-key', same error.
```
root@ubnt# set interfaces wireguard wg0 private-key "oE1fQ0FXtYkMYH9eRttE0pCDeX+IRa711l8ZOm2K8lo="
[edit]
root@ubnt# commit
[ interfaces wireguard wg0 private-key oE1fQ0FXtYkMYH9eRttE0pCDeX+IRa711l8ZOm2K8lo= ]
fopen: No such file or directory

Commit failed
```

@julianschuh you should configure like:

```
sudo su
wg genkey > /config/auth/wg.private

configure 
...
set interfaces wireguard wg0 private-key /config/auth/wg.private
commit 
save
```

**Comment by Lochnair** [comment]
@sskaje 
Yes I know. It should be possible to specify the private key directly on the command line, but for some reason that doesn't work on the ER-X, on my ER-Lite however it works perfectly fine.

**Comment by Lochnair** [comment]
This turned out to be a bug in the ER-X firmware. We use Bash process substitution to provided an in-config private key to the wg util, but process substitution doesn't work on the ER-X with current firmware.

This is supposedly fixed in 1.10.0-alpha1, so I'd advise testing the alpha firmware if possible.

**Comment by Lochnair** [comment]
This got fixed in #27, and is included in the latest release.
